pkgs/by-name/gr/gradle-completion/package.nix

```diff
@@ -1,37 +1,51 @@
-{ lib, stdenv, fetchFromGitHub }:
-
-stdenv.mkDerivation rec {
+{
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  installShellFiles,
+  gitUpdater,
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "gradle-completion";
   version = "1.4.1";
 
   src = fetchFromGitHub {
     owner = "gradle";
     repo = "gradle-completion";
-    rev = "v${version}";
+    rev = "v${finalAttrs.version}";
     sha256 = "15b0692i3h8h7b95465b2aw9qf5qjmjag5n62347l8yl7zbhv3l2";
   };
 
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+
   strictDeps = true;
 
   # we just move two files into $out,
   # this shouldn't bother Hydra.
   preferLocalBuild = true;
 
   dontBuild = true;
+
   installPhase = ''
     runHook preInstall
 
-    mkdir -p $out
-    install -Dm0644 ./_gradle $out/share/zsh/site-functions/_gradle
-    install -Dm0644 ./gradle-completion.bash $out/share/bash-completion/completions/gradle
+    installShellCompletion --bash --name gradle gradle-completion.bash
+    installShellCompletion --zsh _gradle
```

Comment:
```suggestion
    installShellCompletion --cnd gradle \
      --bash gradle-completion.bash \
      --zsh _gradle
```

Reply:
Good catch, I forgot that the script is pretty smart 🙂

I didn't use this verbatim, but I did something very similar:

https://github.com/NixOS/nixpkgs/blob/c4c5c9049ef5a7e6cd1b0f2df674e76001f757c8/pkgs/by-name/gr/gradle-completion/package.nix#L34-L40